lib/src/main/java/com/cloudinary/android/preprocess/ImagePreprocessChain.java

```diff
@@ -24,6 +25,18 @@ public static ImagePreprocessChain limitDimensionsChain(int maxWidth, int maxHei
                 .addStep(new Limit(maxWidth, maxHeight));
     }
 
+    /**
+     * Convenience method for building an upload widget preprocessing chain.
+     * Use this in {@link com.cloudinary.android.UploadRequest#preprocess(PreprocessChain)}.
+     * @param result The result from the upload widget.
+     * @return The prepared chain to pass on to {@link com.cloudinary.android.UploadRequest#preprocess(PreprocessChain)}
+     */
+    public static ImagePreprocessChain uploadWidgetChain(UploadWidget.Result result) {
+        return (ImagePreprocessChain) new ImagePreprocessChain()
+                .addStep(new Rotate(result.getRotationAngle()))
```

Comment:
What happens when the user didn't rotate or crop (or neither?). Rotation may work since int is initiated to zero, but crop points will be null - you create the chain here with invalid values? Or is this taken care of elsewhere?

Reply:
The crop points are being initialized at the ImageView, having a default value of the image size. At the cropping preprocess level - If there is no change to the bitmap size, the original bitmap is returned.